examples/nvram/store.c

```diff
@@ -0,0 +1,170 @@
+/* store.c
+ *
+ * Copyright (C) 2006-2021 wolfSSL Inc.
+ *
+ * This file is part of wolfTPM.
+ *
+ * wolfTPM is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfTPM is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+/* Tool and example for storing a TPM key into the TPM's NVRAM
+ *
+ * NB: This example uses Parameter Encryption to protect the password of the
+ *     TPM NVRAM Index, where the private and public parts of a TPM key is stored
+ *
+ **/
+
+#include <wolftpm/tpm2_wrap.h>
+
+#include <examples/nvram/store.h>
+#include <examples/tpm_io.h>
+#include <examples/tpm_test.h>
+#include <examples/tpm_test_keys.h>
+
+#include <stdio.h>
+
+#ifndef WOLFTPM2_NO_WRAPPER
+
+/******************************************************************************/
+/* --- BEGIN TPM Keygen Example -- */
+/******************************************************************************/
+static void usage(void)
+{
+    printf("Expected usage:\n");
+    printf("./examples/nvram/store [filename] [-aes/-xor]\n");
+    printf("* filename: point to a file containing a TPM key\n");
+    printf("\tIf not supplied, default filename is \"keyblob.bin\"\n");
+    printf("* -aes/xor: Use Parameter Encryption\n");
+}
+
+int TPM2_NVRAM_Store_Example(void* userCtx, int argc, char *argv[])
+{
+    int rc;
+    WOLFTPM2_DEV dev;
+    WOLFTPM2_KEYBLOB keyBlob;
+    WOLFTPM2_SESSION tpmSession;
+    WOLFTPM2_HANDLE parent;
+    WOLFTPM2_NV nv;
+    TPM2B_AUTH auth;
+    word32 nvAttributes;
+    const char* filename = "keyblob.bin";
```

Comment:
Can we make this a command line argument? Or followup with another PR?

Reply:
Let's follow-up with another PR for NVRAM examples enhancements

Reply:
nvram/store already accepts filename, but nvram/read does not. I will make sure to add that.